usage/rest-client/src/main/java/org/apache/brooklyn/rest/client/BrooklynApi.java

```diff
@@ -114,11 +150,23 @@ public BrooklynApi(String endpoint, @Nullable Credentials credentials) {
         }
     }
 
+    /**
+     * @deprecated since 0.9.0. Use {@link BrooklynApiFactory#getBrooklynApi(java.net.URL, String, String)} instead
+     * @see org.apache.brooklyn.rest.client.DefaultBrooklynApiFactory
+     * @see org.apache.brooklyn.rest.client.PoolingBrooklynApiFactory
+     */
+    @Deprecated
```

Comment:
I heartily agree the factory is just a means to an end, in this case easy configurability (independence from any particular JAX-RS implementation would be another, but it's not the point of this PR).
We can achieve that goal in many ways! I'll do an alternative implementation with the static pool as @ahgittin suggested. ^_^